src/main/java/de/tum/in/www1/artemis/repository/TextExerciseRepository.java

```diff
@@ -47,21 +59,17 @@ default TextExercise findByIdElseThrow(long exerciseId) {
         return findById(exerciseId).orElseThrow(() -> new EntityNotFoundException("Text Exercise", exerciseId));
     }
 
-    @Query("""
-            SELECT DISTINCT e FROM TextExercise e
-            LEFT JOIN FETCH e.gradingCriteria
-            WHERE e.id = :exerciseId
-            """)
-    Optional<TextExercise> findByIdWithGradingCriteria(long exerciseId);
+    @EntityGraph(type = LOAD, attributePaths = { "gradingCriteria" })
+    Optional<TextExercise> findWithGradingCriteriaById(long exerciseId);
```

Comment:
Is `@Param` not necessary? You added it somewhere in the other PRs

Reply:
According to our guidelines, it's required when using @Query. We don't have an explicit query here, so it's fine.